### PR TITLE
Get user obligations for reserve logic

### PR DESCRIPTION
**Pull request by mihalex98**
Get user obligations for reserve logic + README

### Checklist

- [] if I'm working with the latest farms.so version, I have modified the so with the latest one from master
- [] if I'm working with the latest kamino_lending.so version, I have modified the so with the latest one from master

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,3 +83,42 @@ $ cp ../kamino-lending/target/deploy/kamino_lending.so deps/programs/kamino_lend
 $ cp ../kamino-lending/target/idl/kamino_lending.json src/idl.json
 $ yarn codegen
 ```
+
+### Getting a vanilla obligation for a user ofr a specific reserve (will return null if the reserve is not used in the obligation)
```
Comment (andreihrs) [inline README.md:87]:
```suggestion
### Getting a vanilla obligation for a user for a specific reserve (will return null if the reserve is not used in the obligation)
```

File: src/classes/obligation.ts [hunk 2]
```diff
@@ -402,10 +402,14 @@ export class KaminoObligation {
       newStats.userTotalBorrowBorrowFactorAdjusted.plus(borrowValueBorrowFactorAdjustedUSD)
     );
 
+    console.log('borrowPosition.amount, borrowAmount', borrowPosition.amount, borrowAmount);
```
Comment (andreihrs) [inline src/classes/obligation.ts:405]: remove logs

File: src/classes/market.ts [hunk 3]
```diff
@@ -781,6 +781,93 @@ export class KaminoMarket {
...
+            }
+          }
+        }
+      });
+    }
+
+    return finalObligations;
+  }
+
+  async getUserVanillaObligationForReserve(user: PublicKey, reserve: PublicKey): Promise<KaminoObligation | null> {
```
Comment (andreihrs) [inline src/classes/market.ts:850]: I think this should rather be a function where you check that the reserve is in the obligation deposits/borrows or not, hence return true or false. 

in your case, you either return the obligation or null. but you can only have 1 vanilla obligation per user in a lending market, so I don’t think it makes sense to “get the user obligation based on the reserve”

File: src/classes/market.ts [hunk 4]
```diff
@@ -781,6 +781,93 @@ export class KaminoMarket {
     });
   }
 
+  async getAllUserObligationsForReserve(user: PublicKey, reserve: PublicKey): Promise<KaminoObligation[]> {
```
Comment (andreihrs) [inline src/classes/market.ts:784]: can you add a test for this?

File: src/classes/market.ts [hunk 5]
```diff
@@ -781,6 +781,93 @@ export class KaminoMarket {
...
+        new LeverageObligation(
+          kaminoReserve[1].getLiquidityMint(),
+          targetReserve.getLiquidityMint(),
+          this.programId
+        ).toPda(this.getAddress(), user)
+      );
+    }
+    const batchSize = 100;
+    const finalObligations: KaminoObligation[] = [];
+    for (let batchStart = 0; batchStart < obligationAddresses.length; batchStart += batchSize) {
```
Comment (andreihrs) [inline src/classes/market.ts:827]: why do you need batches?

**Review by andreihrs — CHANGES_REQUESTED** [review]

File: src/classes/market.ts (hunk 3 above)
Comment (mihalex98) [reply src/classes/market.ts:850]: refractored and split into 2 separate functions

File: src/classes/market.ts [hunk 6]
```diff
@@ -781,6 +781,93 @@ export class KaminoMarket {
...
+    }
+    for (const kaminoReserve of this.reserves) {
+      if (targetReserve.address.equals(kaminoReserve[0])) {
+        // skip target reserve
+        continue;
+      }
+      obligationAddresses.push(
+        new MultiplyObligation(
+          targetReserve.getLiquidityMint(),
+          kaminoReserve[1].getLiquidityMint(),
```
[the remark this answers is not in the document]
Comment (mihalex98) [reply src/classes/market.ts:799]: It worked as the retuned object while looping a map with a size 2 array of key and value. However, I am defining them more clearly in the for loop head now.

File: src/classes/market.ts (hunk 4 above)
Comment (mihalex98) [reply src/classes/market.ts:784]: I've tested using the CLI and it worked. I've now added a test as well

File: src/classes/obligation.ts [hunk 7]
```diff
@@ -406,6 +406,8 @@ export class KaminoObligation {
     borrowPosition.mintAddress = mint;
     borrowPosition.marketValueRefreshed = positiveOrZero(borrowPosition.marketValueRefreshed.plus(borrowValueUSD));
 
+    console.log('newStats', newStats);
```
Comment (andreihrs) [inline src/classes/obligation.ts:409]: remove plis

File: README.md [hunk 8]
```diff
@@ -83,3 +83,45 @@ $ cp ../kamino-lending/target/deploy/kamino_lending.so deps/programs/kamino_lend
 $ cp ../kamino-lending/target/idl/kamino_lending.json src/idl.json
 $ yarn codegen
 ```
+
+### Getting a vanilla obligation for a user
+```ts
+  const kaminoMarket = await KaminoMarket.load(env.provider.connection, marketAddress, DEFAULT_RECENT_SLOT_DURATION_MS, programId);
+
+  const obligation = await kaminoMarket!.getUserVanillaObligationForReserve(user);
```
Comment (andreihrs) [inline README.md:91]: readme is deprecated it seems

**Review by andreihrs — APPROVED** [review]

**Pull request merged**